src/App/VPSScan/Scan/Core.hs

```diff
@@ -0,0 +1,95 @@
+module App.VPSScan.Scan.Core
+  ( coreAuthHeader
+  , createCoreProject
+  , completeCoreProject
+  , getSherlockInfo
+  , createLocator
+  , createRevisionLocator
+  , buildRevision
+  , SherlockInfo(..)
+  )
+where
+
+import App.VPSScan.Types
+import App.Util (parseUri)
+import Data.Text (pack, Text)
+import Prelude
+import Network.HTTP.Req
+import Data.Text.Encoding (encodeUtf8)
+import Control.Monad.IO.Class (liftIO, MonadIO)
+import Control.Effect.Diagnostics
+import Data.Aeson
+import Data.Time.Clock.POSIX (getPOSIXTime)
+
+data SherlockInfo = SherlockInfo
+  { sherlockUrl :: Text
+  , sherlockClientToken :: Text
+  , sherlockClientId :: Text
+  , sherlockOrgId :: Text
+  }
+  deriving (Eq, Ord, Show)
+
+instance FromJSON SherlockInfo where
+  parseJSON = withObject "" $ \obj -> do
+    auth <- obj .: "auth"
+    sherlockUrl <- obj .: "url"
+    sherlockOrgId <- obj .: "orgId"
+    sherlockClientId <- auth .: "clientId"
+    sherlockClientToken <- auth .: "clientToken"
+    return (SherlockInfo { sherlockUrl = sherlockUrl, sherlockClientId = sherlockClientId, sherlockClientToken = sherlockClientToken, sherlockOrgId = sherlockOrgId })
+
+coreAuthHeader :: Text -> Option scheme
+coreAuthHeader apiKey = header "Authorization" (encodeUtf8 ("Bearer " <> apiKey))
+
+
+buildRevision :: (MonadIO m) => Text -> m Text
+buildRevision "" = do
+  posixTime <- liftIO getPOSIXTime
+  pure (pack $ show $ (floor $ toRational posixTime :: Int))
+buildRevision userProvidedRevision = pure (userProvidedRevision)
+
+createLocator :: Text -> Text -> Text
```

Comment:
I'm curious if our `Locator` type in `Srclib.Types` could be reused here

Reply:
It doesn't include explicit support for `orgId`, we'd have to manually build it in to the `locatorProject` string.
But I do agree we shouldn't use `Text`- `newtype`'d it.